### ISSUE TITLE
Pitch instability

### ISSUE THREAD

**Issue by wlbragg** [c172p-team/c172p#1414]
Per https://sourceforge.net/p/flightgear/codetickets/2803/

>The current c172 seems to have pitch instability (tends to hunt up and down, even when properly trimmed). This may be a contibuting factor to takeoff stalls. Movement of cg forward from 39.06
to 37.06 ameliorated the problem.
Also elevator response seemed abrupt. Changing flight_control name="FCS: c172
from <time>0.15</time>[code]to <time>0.30</time>
reduced the abruptness.
Kindly review.
Thanks

@dany93 would you mind reviewing this if you have the time?

**Comment by dany93** [comment]
The CG location has not been determined by trial ans error, but from the respective POHs.

**160hp 1982**:
https://topsflyingclub.files.wordpress.com/2012/09/6508k-poh.pdf
https://fr.scribd.com/document/573746293/Cessna-172P-POH#
PDF 55/160 (p.6-9), others: 105-358 (p.6-9), or 112/357 (p.6-10).
Empty Weight 1467 lb
Moment 57300 lb.ins
Which gives Arm = 57300/1467 = **39.06 in**.

180hp-552SP
http://www.engineersflyingclub.org/planes/552SPPOH.pdf
PDF 156/384 (p.6-12)
Empty Weight 1642 lb
Moment 62600 lb.ins
Which gives Arm = 62600/1642 = **38.12 in**.

Each one (160 hp and 180 hp) is exactly what we have in our FG C172 when empty.

**Pitch instabilities**:
If properly trimmed, I can have it rather stable IMO:
160hp with copilot, 1712 lb, CG 39.07 in
1500 - 2000 ft
2380 rpm
112 kt
aileron trim 0.016 (not the issue, only for comfort and stability)
elevator trim 0.139
For horizontal flight (close to 0 fpm climb rate) : +/- 100 ft oscillations
pitch deg -0.4 ... -1 

Observed during several minutes, for stabilization. This weak oscillations seem normal.

**Tendency to pitch up after takeoff (at default load)**:
Elevator trim has to be preset before takeoff, depending on load. True for every aircraft.
Default: elevator trim = 0. The aircraft takes off on its own close to 60 kt, with no stall tendency just after takeoff. Pitch up tendency later at climbing: must be corrected.
Other possible choice at default load (for climbing): elevator trim = 0.04 prior to takeoff.
With pilot alone (default):  it climbs at 80 kt with weak elevator corrections if elevator trim is set at about 0.04.
With a copilot, it climbs at 80 kt with weak elevator corrections keeping elevator trim = 0.


(Frank Galloney wrote)

> Also elevator response seemed abrupt. Changing flight_control name="FCS: c172
> from 0.15[code]to 0.30
> reduced the abruptness.

Why not ? It seems reasonable. I don't mind. We can try this.

**Comment by wlbragg** [comment]
Thanks @dany93

>Why not ? It seems reasonable. I don't mind. We can try this.

I tried the ..30 value and I felt like it did smooth it out a bit and gave a better feel. My analysis doesn't really mean much however, only that I enjoyed the feel of the aircraft more than when set to .15.

I can make the change and push a merge.

I think the cg change is problematic. It is subjective and beings we're using POH guidelines, it's probably best to leave it alone. However, I wouldn't be opposed to adding a means to adjust CG, like something to mimic luggage location. One interesting way would be a couple hotspots in the cabin to click on to add weight to that location, to in effect "trim" CG to a personal preference. I have heard of pilots arranging luggage for that purpose.

**Comment by dany93** [comment]
Hi @wlbragg 
> I tried the ..30 value and I felt like it did smooth...
> I can make the change and push a merge.

If you can, thank you.

> However, I wouldn't be opposed to adding a means to adjust CG, like something to mimic luggage location

Do as you prefer, I trust you. Your method destroys nothing.

Another tempting and less artificial way would be changing the AERORP (more aft).
In JSBSim the AERORP is the the resulting point where the aerodynamic forces are applied.
Thus and from our 'pitch down' point of view, moving the AERORP backwards has the same effect as moving the FG frontwards.
Currently,
```
    <metrics>
        (...)
        <location name="AERORP" unit="IN">
            <x> 40.6 </x>
```
That's a difficult problem, but I usually consider that the (JSBSim) AERORP is the aerodynamic center of the wing. And the aerodynamic center of a wing is usually said to be close to 25% of the (mean aerodynamic) chord. Which, for the C172, makes 58.8*0.25=14.7 from the leading edge.
The leading edge is at 25.90. This makes AERORP at 25.9+14.7=40.6.

But the aerodynamic center position of a wing is also said to be 23-27% of the mean chord.
https://www1.grc.nasa.gov/beginners-guide-to-aeronautics/aerodynamic-center/
@okroth Pitch axis too sensitive https://github.com/c172p-team/c172p/issues/1047#issuecomment-382449833
Which could make 25.9+58.8*0.27=41.8. I don't know if that's enough for our issue.
I have to try this. If we locate the AERORP too much aft, the risk is that the elevator will not be efficient enough to reach the stall AoA in the most forward CG locations. Consequently, we may have to increase the elevator efficiency, with the risk of some more abruptness.

**Comment by wlbragg** [comment]
Interesting. I tried shifting weight around by fuel and passenger and an added pointmass simulating  putting something with weight forward in the cabin. The best I could do to attempt to get to 37.06 was to get to 38.23. That was adding 300 lbs pilot and co pilot, duel wing tank option with less than a gal. of fuel and adding a point mass of 75 lbs about 24" forward of the pilot/copilot position, as if you added some weight under the dashboard at the passengers feet.

I may be wasting time here by going off on a wild goose chase, but one thing I think I should investigate though is the fuel tanks. I think the exact position of the tanks and where the fuel sets in them could make a great deal of difference as they are sitting forward of CG. 
Look at the location of 39.06in/0.992124m the default CG with default fuel and the pilot.
https://i.imgur.com/zPxkL5v.jpg
Looking at this, doesn't it seem that adding fuel should move CG forward instead of aft? It does the opposite, adding fuel shifts CG aft.

AERORP, so 23-27% is the range and your saying that if we stick within that range, at the aft end of it, we can have the same effect as moving CG forward. Does it actually change CG at all?

**Comment by wlbragg** [comment]
I tried the AERORP change and have to admit, with the two changes FCS: c172 from 0.15 to 0.30 and the AERORP change. The aircraft feels really stable and a joy to fly.
Using the dual tanks, limited fuel, and the 180 engine as well as 2 300 lbs people in the front seats, CG was at 37.56. But even with a more normal load it was at 38.78 and still felt stable.
I'm not sure about your concern with the stall angle or speed, I need more info from you on what to look for or details on what the ramifications of these changes will do to it. I have no understanding of it.

**Comment by dany93** [comment]
@wlbragg wrote

> AERORP, so 23-27% is the range and your saying that if we stick within that range, at the aft end of it, we can have the same effect as moving CG forward. Does it actually change CG at all?

Not at all.
The AERORP location is the result of the aerodynamic forces. For us, the result of calculations ans experimental observations. We have to take a value. But we can presume that there is an uncertainty on this value.
Let's consider the same aircraft. Same CG.
If the AERORP chosen is backward, the CG in forward from it. The aircraft, like hung at its AERORP, will pitch down.

> Looking at this, doesn't it seem that adding fuel should move CG forward instead of aft? It does the opposite, adding fuel shifts CG aft.

The empty CG is at 39.06.
The fuel tanks are at 48: aft from the empty CG. Adding fuel will move the CG backward. Or do I miss something?

**Comment by wlbragg** [comment]
Yeah, I think we are using a bogus position for tanks. They should be forward of CG, not aft.

![Capture](https://user-images.githubusercontent.com/11069680/235370450-c07ec0ad-0439-4931-9a41-fe7e2e6f16ea.JPG)

Sorry for the faint CG and FDM tank positions.
In this image they are at
FDM Tanks x=48in/1.21m
Actual Tanks should be closer to x=12.08in/0.30m
CG x=39.06in/0.99m

**Comment by wlbragg** [comment]
Leaving the original AERORP alone and changing the Fuel tanks to 12.08.

With 40 lbs of fuel in each tank and a 160 lbs pilot changes the CG to 37.51.

Full fuel tanks and two front seat people at 160 lbs each brings CG to 35.17

Add two back seat 160 lb passengers and 50 lbs of luggage brings CG to 41

Empty weight is still 39.06 CG.

Adding fuel to the tanks moves CG forward as I think it should.

**Comment by wlbragg** [comment]
Everything I read indicated adding fuel does shift CG slightly aft. Unless it is the aux tank located behind the rear seats, in which case it shifts it more aft because it is farther away from CG
But with this then I have to surmise the "empty" CG has to be at the mid to, slightly forward of mid, "wing tanks" in order for adding fuel to change CG aft. If CG is aft of wing tanks as we show in the Blender model, then adding fuel has to move cg forward. I think we have some adjusting to do to "empty" CG if we are to match the prevailing documentation. it needs to be moved forward. But you state we get the CG from the POH. So is our model out of position and/or the AERORP or some other factor?

**Comment by wlbragg** [comment]
P.S. The 48in position of the FDM fuel tank could be accounting for the AUX tank mentioned that is behind the rear seat. But the main tanks are not and look to not be accounted for. The 48in position could not be on the wings as it is to far aft of them.

Again, unless all these position in the FDM are not representative of the physical model in Blender.

**Comment by wlbragg** [comment]
Here are the exact parameters.
![Capture](https://user-images.githubusercontent.com/11069680/235379246-d2cae14d-856c-41bc-bf31-001b7a073be8.JPG)

**Comment by dany93** [comment]
@wlbragg wrote
> Again, unless all these position in the FDM are not representative of the physical model in Blender.

Your issue is here.
The Blender axes are not the POH axes.
The POH reference is the Datum. It is also the FDM reference.
![C172S-POH-DATUM](https://user-images.githubusercontent.com/11994061/235430220-765aff71-ef90-43dc-b1f8-8b157555d32a.png)
In Blender, the POH Datum is at about X=-089 m, 0.89 m forward (for the aircraft) the Blender origin.
![image](https://user-images.githubusercontent.com/11994061/235431479-94a0fd29-5a85-4675-b7c4-8391af150f57.png)

To check this, notice the propeller or wheel locations in Blender, and compare with the FDM values.

To make observations easier, I've made a Blender file with a 0.89 m shift. Such as the POH Datum is coincident with the Blender origin.
![image](https://user-images.githubusercontent.com/11994061/235432745-d9afde5f-b74d-4bb8-821f-b05a8aa60049.png)

**Comment by dany93** [comment]
@wlbragg wrote
> I'm not sure about your concern with the stall angle or speed, I need more info from you on what to look for or details on what the ramifications of these changes will do to it. I have no understanding of it.

I do the stall tests:
- Either at horizontal flight, engine at idle, trying to keep the altitude by progressively pulling the yoke.
The airspeed decreases, the AoA increases (pitch up) up to the stall.
- Or in a steep turn, with the engine at high regime.

To reach the stall AoA, the elevator must have enough efficiency. Otherwise, you are with yoke fully pulled off, and the aircraft goes on "flying", not stalled. You never reach down to the stall airspeed. Maybe it loses altitude, but not stalled.
Consequence: at landing, you cannot reach the minimum airspeed, you can only touch the ground at a greater airspeed. Which is not that good for the gears, and means that you will need a longer distance to stop.

All of this to say that you need enough efficiency from the elevator (pitch up) to reach the stall.
If the CG is too forward, the aircraft will tend to pitch down, you will constantly have to pull the yoke even in flying, and you will no longer be able to reach the stall angle and stall airspeed. 

Is it enough?

**Comment by dany93** [comment]
I had a doubt...

If you measure visually in Blender, you find the wing leading edge at X=29 (Blender) in instead of 25.90 in (POH).
Which might be confusing if you measure (evaluate?) the tank locations in Blender.

However, the tank locations, X = 48 in the FDM, are accordance with the POHs.
160 hp [PDF 112/357, p.6-10] 11600 lbs.ins / 240 lbs = 48.3 in
180 hp [PDF 156/384, p.6-12] 8600 lbs.ins / 180 lbs = 47.8 in.

Also, I have checked the CG at max load in FG for our 160 hp and 180 hp. I see close to 45 and 44 ins with luggage. Which is in accordance with the same calculation (moment / weight) in the tables for "TAKE OFF WEIGHT AND MOMENT".
160 hp 107800 / 2400 = 44.9
180 hp 112800 / 2550 = 44.24

**Comment by dany93** [comment]
Hi @wlbragg,

In the FDM lines 1703-... there is a Pitching_moment_at_zero_alpha. It adds a pitch up moment (depending on airspeed) to the moment due to elevator deflection.
The current values are taken from Megginson's table. These are calculated (not wind tunnel).
By changing like this:
```
        <axis name="PITCH">
            <function name="aero/coefficient/Cmo">
                <description>Pitching_moment_at_zero_alpha</description>
                       skip(....)
                    <table>
                        <independentVar lookup="row">aero/qbar-psf</independentVar>
                        <!-- <tableData>
                            13.6  0.09
                            21.2  0.04
                        </tableData> -->
                        <tableData>
                            13.6  0.07
                            21.2  0.02
                        </tableData>
                    </table>
                </product>
            </function>
```
This pitch up moment is decreased.
Do you think it's better?
Not easy, we don't know the real C172, and these tests have to be done at least at low, with CG forward, and high loads with CG backwards.

**Comment by wlbragg** [comment]
@dany93 

>To make observations easier, I've made a Blender file with a 0.89 m shift. Such as the POH Datum is coincident with the Blender origin.

I think you may have explained this to me before. I forget. IMHO It really complicates everything from the point the decision was made by the original FDM developer to do it this way. Or I am still not totally understanding. Having to account for offsets at every modeling step is tedious at best. With my limited knowledge it seems like a poor decision to start modeling with that handicap. I know it was a real headache when positioning contact points and all the gear changes. Nothing ever lined up and at the time I worked on those features I wasn't as knowledgeable as my "still" limited knowledge is now.

>I've just discovered that we can display the dimensions directly in inches in Blender.

Yeah, I knew about it, I'm just too lazy to find the setting switch again and change it, ha!

I'll test this in the next couple days. Up till now, with any of the changes discussed, I haven't had any issue with stall. Although I think my testing so far was only power on stall, if that makes any difference.

I wanted to ask, you must feel at least somewhat enthusiastic about these suggested changes, or if not the changes themselves, the resulting difference made to the feel of the FDM, or you wouldn't be working this. Is this correct? I'm not interested in making changes for the sake of making changes and I don't think you are either. However I do value input from current pilots of the aircraft and it does seem like one of the main themes we keep hearing about is "stability". So with that in mine I feel like this is a worthwhile endeavor.

Side note, I have a couple other minor changes I want to make issues for and then merge. Plus there are a few other issues here on GitHub I want to finish up. Once this is all finished, I'll cut a new Version tag. It's been too long since we have made one.

**Comment by dany93** [comment]
The sketches above are instructive.
What I notice is:
- The propeller axis is at about Z=30 inches above, (rather 31) from the FDM axis. Instead of 26.6 inches in the current git FDM version.
A propeller which is **lower** than the general aircraft / wings axis gives a **pitch up moment**, and vice-versa. (more rigorously, the height reference should be the aerodynamic center of pressure of the whole aircraft)
Changing this value for a **higher** propeller location would **decrease** the **general pitch up moment** at full thrust. 
Changing from 26.6 to 30 or 31 would only have a weak effect, but in the right direction for us.
- The AERORP:
[JSBSim Aerodynamics : metrics](https://wiki.flightgear.org/JSBSim_Aerodynamics#Metrics)

> "AeroRP - the location where the aerodynamic forces are applied to the model. If this location is not exactly equal to the center of gravity (CoG) then moments, or twisting forces, will be created."

This is a subject that I have always found difficult to understand completely. I think I've made some progress.
My current understanding:
The AERORP can be located anywhere, the aerodynamic forces will be applied here.
But the moment by each of these forces is not always automatically known by the FDM. For the aerodynamic forces, the arms must be given for the FDM being able to calculate it.
In the C172P, I notice that the AERORP is at Z=59.4 (at the wings). With this location as a reference for the calculation of the moments, the propeller at full thrust gives a rather strong pitching moment. Which, I think, is normally compensated by the pitch down moment from the fuselage drag. But, in the current C172P, I see no such pitch down moment due to fuselage drag. I'm surprised that no one noticed this since the time this aircraft exists,  I hope not having misunderstood something.
It could be fixed by two ways:
1) Letting the AERORP where it is, give an arm to JSBSim and calculate the fuselage drag moment. This arm is to be evaluated (distance between the AERORP and the aerodynamic center of the fuselage + wings). Not easy.
or
2) Placing the AERORP at the height of the propeller axis, which can be presumed not far from the fuselage aerodynamic center. Which minimizes the moments to be calculated, thus the errors. I prefer this one.

I will look at this more, and (I hope) make a pull request for the FDM with these ideas. I hope it will noticeably decrease the general pitch up tendency of our C172.

**Comment by dany93** [comment]
On 12 may 2023 in FGDevel mailing list, James Turner wrote:

> Are there any changes / fixes on next, which should be back ported to 2020.3 branch, assuming I will make a new stable release in the coming week?

Is our current C172 on git next branch compatible with FG 2020.3?

**Comment by wlbragg** [comment]
>Is our current C172 on git next branch compatible with FG 2020.3?

I'm not sure, I'd have to take a closer look. I can install 2020.3 and see. I just don't know exactly when. Maybe I can answer that tonight.

As far as your above examination, I'll read in in detail shortly, I'm really short of time at the moment. But there is so much here I want to get done, soon!

**Comment by dany93** [comment]
@wlbragg wrote

> I'm really short of time at the moment

As you have too much time left :smile: , this Blender screen copy. Shifted and reoriented such that the FDM and POH (X DATUM) axes are coincident and consistent with the FDM axes. As well as I could, correctly I hope. Very handy for observing and measuring locations.
![C172-axes-POH-FDM-mesures](https://github.com/c172p-team/c172p/assets/11994061/e0aad19d-6a59-4aa8-8403-076a4ebadb3e)

I'm tempted by adding this Blender file (with inches units) in the dev/blender directory of the git version (about 4.7 MB added). What do you think of it?

**Comment by dany93** [comment]
@wlbragg 
Pull request.
The tendency to pitch up is clearly reduced. At climbing after takeoff, I only need a bit of elevator pitch trim (down) at full load and with rear CoG (44 inches). This seems normal. The choice is a compromise.

**Comment by wlbragg** [comment]
>I'm tempted by adding this Blender file (with inches units) in the dev/blender directory of the git version (about 4.7 MB added). What do you think of it?

Absolutely. go ahead and add it.

>Placing the AERORP at the height of the propeller axis, which can be presumed not far from the fuselage aerodynamic center. Which minimizes the moments to be calculated, thus the errors. I prefer this one.

If this is your prefered solution, as I understand it, I'd like to try this after you make this change. 

I'm having to ask on the mailing list how to get a compiled 2020.3-release version of FG/SG/OSG. I keep running into a "can't find Boost failure".

**Comment by dany93** [comment]
I have corrected the location of VRP X from 42.6 to 35.04.
35.04 is the offset between the 3D model (Blender) origin and the FDM.
This solves an inconsistency that I was trying to resolve for a long time.
I hope I'm not wrong, this is a tricky subject...

I checked by looking at the "jump" of the wheels at the edge of a terrain change. It's OK.

**Comment by dany93** [comment]
@wlbragg 
The Z values for the BOGEY (s), initially set in the FDM, are overwritten later. I can't find how and from where.

**Comment by dany93** [comment]
The initial Z offsets were done by the FDM (VRP) and c172-common.xml. Which was confusing.
Unless I've not understood something (please help me...) this was useless, rather awkward.

I have combined the two Z offsets into a single one, by the FDM VRP only. https://github.com/c172p-team/c172p/commit/248411d6101f3799eeeb908780ef60211235b900
Views: I had to re adjust the Z values for pilots and passengers (I hope with no errors, I just discovered this).

**Comment by wlbragg** [comment]
@dany93

>Otherwise with the current version (AERORP X=41.2), you have to pull the yoke during climbing after takeoff at default load. Or to set the elevator trim up. Do you think this is abnormal?

I wouldn't know! But it sounds reasonable. Let's ask the question in the forum under the c172p detailed thread. I'll do that now. Without going into any explanation on that thread, I'll just ask the basic question.
https://forum.flightgear.org/viewtopic.php?f=4&t=25157&p=412463#p412463

>I may have understood: it is to show the wheel embedded in the grass.

Without looking closely, yes, that sounds right. I do sink the wheels in soft ground and snow. They also are changed on every bushkit changeout, "tire size", undercarriage type.

**Comment by dany93** [comment]
Thank you and good idea @wlbragg for having posted this question in the forum. It brought an interesting response from xDraconian.

**Comment by dany93** [comment]
I have softened the ski spring coefficient for the main bogeys (aft of skis).
It was too hard. The ski bogeys bungees or springs are just to hold the skis in place during flying, not to sustain the aircraft weight.
The consequence was that it prevented the nose from pitching up when taxiing for takeoff. The yoke had to be pulled back largely, and when the aircraft did pitch up, it was brutal.

I will have a few other corrections on wheel locations when this issue is merged, but I don't want to mix everything. This one was more important.

**Comment by wlbragg** [comment]
>I have softened the ski spring coefficient for the main bogeys (small wheels aft of skis).

Cool!

I finally tested our master dev version on LTS 2020.3.18 and it works fine, I was pretty sure it would. I can't think of anything in it that is not compatible. What we probably should do is close out a few of these current outstanding issues, and a couple I have that I want to finish. Then cut a new release to match next and ask to have either that release or make one specifically for LTS 2020.3.18 and have it backported to LTS 2020.3.18. There has been a ton of improvements that would be nice to have in LTS from here on out.
The only thing that actually would need to be edited is the minimum-fg-version tag in c172p-main.xml It is set to current next (2020.4.0) and gives a warning on 2020.3.18.

**Comment by dany93** [comment]
>I finally tested our master dev version on LTS 2020.3.18 and it works fine

Good news!

As I wrote [above](https://github.com/c172p-team/c172p/issues/1414#issuecomment-1574828297), I still have a few features to add.
- A few corrections on wheel locations (easier now with the Blender FDM axes),
- Add two Blender files, with the FDM axes too, for the amphibious and skis. If you think these extra Mo it can be useful.

My first intention was to wait for this issue to be merged. Do you think I can add them now inside this issue, prior to this merging?

**Comment by wlbragg** [comment]
Yeah, go ahead. If they were major stuff a new issue might be better. But they fit here well enough. After all, at least part of it is pitch related.

**Comment by dany93** [comment]
@wlbragg 
Done.
-  Wheel locations (mainly X locations),
- Two Blender files, with the FDM / DATUM axes for the amphibious and skis.

For the amphibious, the wheels (particularly the aft ski wheels) are seen a bit sunk in the hard ground. I've not corrected this for fear of degrading your snow settings. If I understand well, this height from the ground is set by the main wheel (36" diameter) Z  value.

**Comment by dany93** [comment]
@wlbragg, please notice:
Reminder: to simplify things and keep just one offset (merged into VRP), I've transferred the Z offset from c172p-common.xml to the VRP in c172p.xml. This is a merging.
**This offset transfer requires the aircraft cache to be cleaned** (delete c172p.xml in aircraft-data). Only once, the first time.
I assume this is due to a value saved in the cache, but I don't know which one. I don't know how to avoid people getting confused the first time.

Otherwise, unless you find something incorrect, for me it's done, it's good to be merged.

**Comment by wlbragg** [comment]
@dany93 OK, thanks. I'll try to test this as soon as possible. After I get this merged I want to review a couple thing I have floating around and try to get some other outstanding issues resolved. Then hopefully cut a long overdue new tag/release.

**Comment by wlbragg** [comment]
@dany93 
I am still somewhat confused by these concepts, but I have read this thread ne a couple times and each time it makes a little more sense to me.

With that said, something else I am trying to figure out, maybe you can help me with this. It appears the AI wake is fixed? I turned it on and my latest build of FG with this version of the 172 does not crash anymore. I can't really tell if I am getting any wake influence though.

But my question is how should this be measured and what numbers would you use for the following?

**Comment by wlbragg** [comment]
These were the original locations, I think I used an correct reference for the current numbers.
If you can figure this out maybe it will help me visualize this better.
~~~
<x>32.0</x>
<y> 0.0</y>
<z>52.5</z>
~~~

**Comment by dany93** [comment]
```
Center of the main wing leading edge -->
                <x>-6.24</x>
                <y> 0.0</y>
                <z>19.11</z>
```
These are in the usual (initial) Blender coordinates.

They are very close to the coordinates the main wing **leading edge** **in the initial Blender coordinates**.
In these coordinates, the POH aircraft (= FDM) DATUM in at about -35".

The vertical axis is close to (about 6" aft from) the leading edge,
The horizontal axis is slightly below (about 2.3") the propeller axis.

**Comment by dany93** [comment]
In the shifted (= POH and FDM) Blender coordinates, I'd rather say
```
<!-- Center of the main wing leading edge -->
                <x>26</x>
                <y> 0.0</y>
                <z>55</z>
```
Not far from yours.

**Comment by dany93** [comment]
I have some difficulty with
```
   <force name="ai-wake" frame="BODY">
            <location unit="IN">
                <!-- Center of the main wing leading edge -->
                <x>-6.24</x>
                <y> 0.0</y>
                <z>19.11</z>
            </location>
            <direction>
```
The BODY frame in the one for physics calculations. For me, it is the one the origin of which is the AERORP.

However, taking into account the brevity and nature of these AI wake interactions, I wonder if and how one can notice the difference.

**Comment by wlbragg** [comment]
Thank you! That was a really good explanation. Now I think maybe I can finally wrap my head around this.

**Comment by dany93** [comment]
@dany93 wrote
> The BODY frame in the one for physics calculations. For me, it is the one the origin of which is the AERORP.

I have doubts....
AI wakes are aerodynamics forces, but the AERORP can be located anywhere. Anywhere at least in principle, because its location can make moment calculations easier or more complicated, more or less accurate.
In [JSBSim Aerodynamics](https://wiki.flightgear.org/JSBSim_Aerodynamics#Frames) and in [JSBSim Reference Manual](https://jsbsim.sourceforge.net/JSBSimReferenceManual.pdf), the BODY frame location is 'coincident with the CoG'. 

I don't know how the AI wakes are calculated in JSBSim. If they are external forces, setting the CoG as the reference makes sense. External forces are a domain that I don't know.

Sorry for adding confusion, for not being able to respond.

**Comment by dany93** [comment]
If the AI wake simulation is correct, I think that a light aircraft in the wake of heavy one (airliner, helicopter) could easily end up upside down.
In the airports, they keep 2 or 3 min between airliners partially because of this.

**Comment by dany93** [comment]
Blender v2.82.7
This the one with Linux Mint 20.3. Is it a problem?

![c172-FDM-Datum-X26-Z55](https://github.com/c172p-team/c172p/assets/11994061/1ef96205-290f-4fd5-932e-fc14cf14e174)

**Comment by wlbragg** [comment]
I can't read it with 2.79. I had so many aircraft blend files that switching to the new blender updates was problematic. But I have no problem installing and using a newer version for this.

**Comment by dany93** [comment]
With my Blender version (2.82) I can open and read c172p Blender files (*.blend) from 2016 and 2018.
After switching to this (new for me) version, I just had to install io_scene_ac3d > 2.8. https://github.com/NikolaiVChr/Blender-AC3D/tree/2.80

**Comment by dany93** [comment]
I'd like you to be able to read it, because it was intended primarily for you. If everything goes well, it enables to read the wheel and other contact coordinates in Blender and set them in the FDM with no calculations. Much easier, all the more with the pitch offset.

**Comment by wlbragg** [comment]
I installed 82, I can read it now, thanks.
